samcli/lib/sync/sync_flow.py

```diff
@@ -90,10 +100,23 @@ def __init__(
 
     def set_up(self) -> None:
         """Clients and other expensives setups should be handled here instead of constructor"""
-        self._session = Session(profile_name=self._deploy_context.profile, region_name=self._deploy_context.region)
+        pass
+
+    def _get_session(self) -> Session:
```

Comment:
Yeah I thought about this too, but then it becomes harder to keep track of fields since I can't use same name for method and also actual field that will keep the real instance. That is why writing a `_get_{field}` clearer than having 2 instance fields. Let me know if there is a better pythonic way of doing this. :)